src/admin.rs

```diff
@@ -77,23 +77,39 @@ async fn show_lists(stream: &mut OwnedWriteHalf, pool: &ConnectionPool) -> Resul
     ])); // but admin tools that work with pgbouncer want this
     res.put(data_row(&vec![
         "free_clients".to_string(),
-        stats["cl_idle"].to_string(),
+        stats
+            .keys()
+            .map(|address_id| stats[&address_id]["cl_idle"])
+            .sum::<i64>()
+            .to_string(),
     ]));
     res.put(data_row(&vec![
         "used_clients".to_string(),
-        stats["cl_active"].to_string(),
+        stats
+            .keys()
+            .map(|address_id| stats[&address_id]["cl_active"])
+            .sum::<i64>()
+            .to_string(),
     ]));
     res.put(data_row(&vec![
         "login_clients".to_string(),
         "0".to_string(),
     ]));
     res.put(data_row(&vec![
         "free_servers".to_string(),
-        stats["sv_idle"].to_string(),
+        stats
+            .keys()
+            .map(|address_id| stats[&address_id]["sv_idle"])
+            .sum::<i64>()
+            .to_string(),
     ]));
     res.put(data_row(&vec![
         "used_servers".to_string(),
-        stats["sv_active"].to_string(),
+        stats
+            .keys()
+            .map(|address_id| stats[&address_id]["sv_active"])
```

Comment:
You are mapping this 4 times. Can you partition it once instead and then access it by the "cl_active" etc key?

Maybe it is fast enough that we dont care.

Reply:
Yeah it's fast enough we don't care and `SHOW LISTS` shouldn't' be called too often, maybe once every 15 seconds or so normally? This is really fast anyway.